pkg/gateway/model_build_targetgroup.go

```diff
@@ -192,23 +193,39 @@ func (t *latticeServiceModelBuildTask) buildTargets(ctx context.Context) error {
 // Build target group for K8S serviceexport object
 func (t *targetGroupModelBuildTask) BuildTargetGroup(ctx context.Context) error {
 	tgName := latticestore.TargetGroupName(t.serviceExport.Name, t.serviceExport.Namespace)
+	var tg *latticemodel.TargetGroup
+	var err error
+	if t.serviceExport.DeletionTimestamp.IsZero() {
+		// build TargetGroup Model for serviceExport creation
+		tg, err = t.buildTargetGroupForServiceExportCreation(ctx, tgName)
+	} else {
+		// build TargetGroup Model for serviceExport deletion
+		tg, err = t.buildTargetGroupForServiceExportDeletion(ctx, tgName)
+	}
+	if err != nil {
+		return err
+	}
+	t.tgByResID[tgName] = tg
+	t.targetGroup = tg
+	return nil
+}
 
+func (t *targetGroupModelBuildTask) buildTargetGroupForServiceExportCreation(ctx context.Context, targetGroupName string) (*latticemodel.TargetGroup, error) {
 	svc := &corev1.Service{}
 	if err := t.client.Get(ctx, k8s.NamespacedName(t.serviceExport), svc); err != nil {
-		// mark there is no serviceexport dependence on the TG
-		t.datastore.SetTargetGroupByServiceExport(tgName, false, false)
-		return fmt.Errorf("error finding corresponding service %v error :%w", k8s.NamespacedName(t.serviceExport), err)
+		t.log.Infof("Error finding corresponding service %v error :%v \n", k8s.NamespacedName(t.serviceExport), err)
```

Comment:
Returning error instead of logging should be our preference per recent discussion, this seems like going backwards.

Reply:
Will change